src/rabbit_channel.erl

```diff
@@ -789,9 +792,17 @@ check_internal_exchange(#exchange{name = Name, internal = true}) ->
 check_internal_exchange(_) ->
     ok.
 
-check_topic_authorisation(#exchange{name = Name, type = topic}, #ch{user = User}, RoutingKey, Permission) ->
+check_topic_authorisation(#exchange{name = Name = #resource{virtual_host = VHost}, type = topic},
+                          #ch{user = User = #user{username = Username}, conn_pid = ConnPid},
+                          RoutingKey,
+                          Permission) ->
     Resource = Name#resource{kind = topic},
-    Context = #{routing_key => RoutingKey},
+    VariableMap = build_topic_variable_map(
+        amqp_connection:info(ConnPid, [amqp_params]),
```

Comment:
as @lukebakken points out in a different thread, this introduces a dependency on the Erlang client.

We can easily replace this call with
a direct [gen_server call](https://github.com/rabbitmq/rabbitmq-erlang-client/blob/master/src/amqp_gen_connection.erl#L87), since the reliance on `amqp_params` to pass around variables is kind of a hack (relies too much on implementation detail knowledge).